src/main/java/net/syscon/elite/service/CourtHearingsService.java

```diff
@@ -0,0 +1,75 @@
+package net.syscon.elite.service;
+
+import lombok.extern.slf4j.Slf4j;
+import net.syscon.elite.api.model.CourtHearing;
+import net.syscon.elite.api.model.PrisonToCourtHearing;
+import net.syscon.elite.repository.jpa.model.AgencyLocation;
+import net.syscon.elite.repository.jpa.model.CourtEvent;
+import net.syscon.elite.repository.jpa.model.EventStatus;
+import net.syscon.elite.repository.jpa.model.EventType;
+import net.syscon.elite.repository.jpa.repository.AgencyLocationRepository;
+import net.syscon.elite.repository.jpa.repository.CourtEventRepository;
+import net.syscon.elite.repository.jpa.repository.OffenderBookingRepository;
+import net.syscon.elite.service.transformers.AgencyTransformer;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.validation.annotation.Validated;
+
+@Service
+@Transactional(readOnly = true)
+@Validated
+@Slf4j
+public class CourtHearingsService {
+
+    private final OffenderBookingRepository offenderBookingRepository;
+
+    private final CourtEventRepository courtEventRepository;
+
+    private final AgencyLocationRepository agencyLocationRepository;
+
+    public CourtHearingsService(final OffenderBookingRepository offenderBookingRepository,
+                                final CourtEventRepository courtEventRepository,
+                                final AgencyLocationRepository agencyLocationRepository) {
+        this.offenderBookingRepository = offenderBookingRepository;
+        this.courtEventRepository = courtEventRepository;
+        this.agencyLocationRepository = agencyLocationRepository;
+    }
+
+    @Transactional
+    public CourtHearing scheduleHearing(final Long bookingId, final PrisonToCourtHearing hearing) {
```

Comment:
Hopefully quite obvious but this is a work in progress.  Not used outside of service layer though.